File: 프로그래머스/2/1844. 게임 맵 최단거리/게임 맵 최단거리.js
function bfs(maps, visited, n, m) {
    const dx = [0, -1, 0, 1];
    const dy = [1, 0, -1, 0];
    
    let queue = [[0, 0, 1]];
    
    while (queue.length) {

        const [x, y, cnt] = queue.shift()
            
        if ( x === n - 1 && y === m - 1) {
            return cnt; 
        }

        for (let i = 0; i < 4; i++) {
            let nx = x + dx[i];
            let ny = y + dy[i];

            if (0 <= nx && nx < n && 0 <= ny && ny < m && !visited[nx][ny] && maps[nx][ny] === 1) {
                queue.push([nx, ny, cnt + 1]);
                visited[nx][ny] = true;
            } 
        }  
        
    }
    return -1
}

function solution(maps) {
    var answer = 0;
    
    const n = maps.length;
    const m = maps[0].length;
    
    let visited = Array.from({ length: n }, () => Array(m).fill(false));
    visited[0][0] = true
    
    answer = bfs(maps, visited, n, m)
        
    return answer;
}

File: 프로그래머스/2/1844. 게임 맵 최단거리/게임 맵 최단거리.py
from collections import deque

def solution(maps):
    answer = -1
    n = len(maps[0])
    m = len(maps)
    
    dx = [0, 1, 0, -1]
    dy = [1, 0, -1, 0]
    
    visited = [[0] * n for _ in range(m)]
    queue = deque([(0, 0, 1)])
    visited[0][0] = 1
    
    while queue:
        x, y, cnt = queue.popleft()
        
        if x == m - 1 and y == n - 1:
            return cnt      
        
        for d in range(4):
            nx, ny = x + dx[d], y + dy[d]
            
            if 0 <= nx < m and 0 <= ny < n and visited[nx][ny] == 0 and maps[nx][ny] == 1:
                visited[nx][ny] = 1
                queue.append((nx, ny, cnt + 1))
                
    return answer